### ISSUE TITLE
Broken with Node.js 5 / dependencies version constraints not strict enough

### ISSUE THREAD

**Issue by triccardi-systran** [tracker1/gulp-header#29]
When using gulp-header with nodejs 5, I get this error:
```
TypeError: through.obj is not a function
    at module.exports (node_modules/gulp-header/index.js:68:18)
```

`package.json` hs not enough strict version constraints for the dependences:
```
  "dependencies": {
    "gulp-util": "*",
    "object-assign": "*",
    "through2": "*",
    "concat-with-sourcemaps": "*"
  },
  "devDependencies": {
    "event-stream": "^3.1.7",
    "gulp": "^3.9.0",
    "mocha": "*",
    "should": "*",
    "vinyl": "*"
  }
```

EDIT: `through2.obj(fn)` is only available since `through2` `0.4.0`. If another dependency requires an older `through2` then npm3 dedupe will happily put the old version in `node_modules`, and `gulp-header` will crash like this. (for me it happened with `progress-stream`).

All dependencies should use semver constraints whenever possible to avoir this issue.

**Comment by bigtimebuddy** [comment]
+1

Same issue here. Other modules with stricter dependencies are pulling an earlier version of **through2** with NPM3. Highly recommend replacing `*` with fuzzy versions `"through2": "^2.0.0"`

**Comment by destroyerofbuilds** [comment]
Looks like that was fixed in #30 

@triccardi-systran can you confirm you're no longer encountering a problem with Node 5, and if you're not, close out this issue?

**Comment by triccardi-systran** [comment]
@destroyerofbuilds I confirm that by patching my `package.json` the following way it now works with nodejs 5:
```patch
-    "gulp-header": "^1.7.1",
+    "gulp-header": "tracker1/gulp-header",
```

I'm just waiting for the fix to be published on npm.

**Comment by tracker1** [comment]
published 1.8.2, fixed an apparent bug in testing...
